### PR TITLE
Bazel PoC

### PR DESCRIPTION
**Pull request by filippobrizzi**

### CODE REVIEW

File: modules/utils/BUILD [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+package(default_visibility = ["//visibility:public"])
+
+# Generate version_impl.h
+genrule(
+    name = "version_impl",
+    outs = ["src/version_impl.h"],
+    cmd = "./$(location generate_version_file.sh) \"$@\"",
+    local = True,
+    tools = ["generate_version_file.sh"],
+)
```
Comment (filippobrizzi) [inline modules/utils/BUILD:16]: remove `show_version` all together

**Comment by johannes-graeter** [comment]
Since this is a huuuuugggge PR I first made a sanity test, just built and run it as described in the read me.
Only small thing is that my docker version was too old and I had to update since docker compose -f file.yaml was not supported yet. Perhaps just give a minimum docker version in the readme. 

Otherwise it worked perfectly! Tooling is cool (running clang tidy, compile commands config).
I was first a little confused running a makefile, since I did not run one with custom commands for around 10 years but I guess it's quite suited for the use case.

Congrats. I will try to make room to look over the 3000 changes if need be...
Best

**Comment by filippobrizzi** [comment]
> Since this is a huuuuugggge PR I first made a sanity test, just built and run it as described in the read me. Only small thing is that my docker version was too old and I had to update since docker compose -f file.yaml was not supported yet. Perhaps just give a minimum docker version in the readme.
> 
> Otherwise it worked perfectly! Tooling is cool (running clang tidy, compile commands config). I was first a little confused running a makefile, since I did not run one with custom commands for around 10 years but I guess it's quite suited for the use case.
> 
> Congrats. I will try to make room to look over the 3000 changes if need be... Best

Thanks for trying it out! I will add the minimum docker version, do not worry about all the changes. I will try to split the PR in two.

**Comment by johannes-graeter** [comment]
> > Since this is a huuuuugggge PR I first made a sanity test, just built and run it as described in the read me. Only small thing is that my docker version was too old and I had to update since docker compose -f file.yaml was not supported yet. Perhaps just give a minimum docker version in the readme.
> > Otherwise it worked perfectly! Tooling is cool (running clang tidy, compile commands config). I was first a little confused running a makefile, since I did not run one with custom commands for around 10 years but I guess it's quite suited for the use case.
> > Congrats. I will try to make room to look over the 3000 changes if need be... Best
> 
> Thanks for trying it out! I will add the minimum docker version, do not worry about all the changes. I will try to split the PR in two.

Thanks a lot, just give note when its ready to review.

**Pull request merged**